BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m

```diff
@@ -189,6 +189,15 @@ - (NSString *)redirectURIString
 #pragma mark - Token Refresh
 - (void)performRefreshTokenGrant:(NSString *)expiredAccessToken withCompletionBlock:(void (^)(BOXOAuth2Session *session, NSError *error))block
 {
+    if (!self.refreshToken || !self.clientID || !self.clientSecret) {
+        if (block) {
+            NSError *error = [NSError errorWithDomain:BOXContentSDKErrorDomain code:BOXContentSDKAPIErrorUnauthorized userInfo:nil];
```

Comment:
- Can we add a message to the userinfo. e.g. NSUnderlyingErrorKey => @"Missing....
- Can we add a test to BOXOauth2SessionTests to ensure that this error is thrown.